### PR TITLE
Bump version to 3.0

### PR DESCRIPTION
**Pull request by penghuo**
Signed-off-by: Peng Huo <penghuo@gmail.com>
 
### Issues Resolved
https://github.com/opensearch-project/opensearch-plugins/issues/142
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by dai-chen — APPROVED** [review]

**Review by joshuali925 — DISMISSED** [review]

**Review by MaxKsyunz — DISMISSED** [review]

**Comment by penghuo** [comment]
IT failed

File: integ-test/src/test/java/org/opensearch/sql/legacy/AggregationIT.java [hunk 1]
```diff
@@ -166,13 +166,13 @@ public void percentileTest() throws IOException {
...
-            0.001);
+            0.6);
     Assert
         .assertEquals(39.0, getDoubleAggregationValue(result, "PERCENTILES(age)", "values", "95.0"),
-            0.001);
+            0.6);
     Assert
         .assertEquals(40.0, getDoubleAggregationValue(result, "PERCENTILES(age)", "values", "99.0"),
-            0.001);
+            0.6);
```
Comment (Yury-Fridlyand) [inline integ-test/src/test/java/org/opensearch/sql/legacy/AggregationIT.java:175]: Why these tests failed? What was changed?

**Review by anirudha — DISMISSED** [review]

**Comment by MaxKsyunz** [comment]
@penghuo Do you know why Gradle on macOS wasn't finding `opensearch.pluginzip` plugin on sonotype?

I wasn't able to build the `bump/3.0` branch until I built OpenSearch locally and published artifacts to Maven local.
Gradle was failing to find `opensearch.pluginzip` plugin.

**Comment by penghuo** [comment]
missing https://ci.opensearch.org/ci/dbc/distribution-build-opensearch/3.0.0/6181/linux/x64/tar/builds/opensearch/plugins/opensearch-ml-3.0.0.0.zip

File: integ-test/src/test/java/org/opensearch/sql/legacy/AggregationIT.java (hunk 1 above)
Comment (penghuo) [reply integ-test/src/test/java/org/opensearch/sql/legacy/AggregationIT.java:175]: [Percentiles are (usually) approximate](https://www.elastic.co/guide/en/elasticsearch/reference/7.10/search-aggregations-metrics-percentile-aggregation.html#search-aggregations-metrics-percentile-aggregation-approximation). I did not dive deep into the core change yet. I am sure it is not becuase of query engine.

**Review by joshuali925 — APPROVED** [review]

**Review by Yury-Fridlyand — DISMISSED** [review]

**Comment by penghuo** [comment]
Test failed because of breaking change in OpenSearch

```
java.lang.AssertionError: aliases version should be present for indices
```

**Review by dai-chen — APPROVED** [review]

**Review by ps48 — APPROVED** [review]

**Pull request merged**